pkg/actuators/machine/instances.go

```diff
@@ -134,3 +134,25 @@ func getNetworkID(network powervsproviderv1.PowerVSResourceReference, client pow
 
 	return nil, fmt.Errorf("failed to find a network ID")
 }
+
+func getServiceInstanceID(serviceInstance powervsproviderv1.PowerVSResourceReference, client powervsclient.Client) (*string, error) {
+	if serviceInstance.ID != nil {
+		klog.Infof("Found the service %v", serviceInstance)
+		return serviceInstance.ID, nil
+	} else if serviceInstance.Name != nil {
+		serviceInstances, err := client.GetCloudServiceInstances()
+		if err != nil {
+			klog.Errorf("failed to get serviceInstances, err: %v", err)
+			return nil, err
+		}
+		for _, si := range serviceInstances {
+			if si.Name == *serviceInstance.Name {
+				klog.Infof("serviceInstance %s found with ID: %s", *serviceInstance.Name, si.Guid)
+				return &si.Guid, nil
+			}
+		}
+		return nil, fmt.Errorf("failed to find a service ID with name %s", *serviceInstance.Name)
```

Comment:
There is a possibility that 2 instances can have same name, we need to through an error message in that case.

Reply:
Added a check to verify against two service instances having same name